### PR TITLE
[5.1] Added seeJsonKeys and dontSeeJsonKeys to CrawlerTrait

### PR DESCRIPTION
**Pull request by samlev**
Added methods to the CrawlerTrait which allow you to search
for json keys without caring about content.

This is useful if returned data is long, or the exact value doesn't
really matter.

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 1]
```diff
@@ -528,6 +528,82 @@ public function seeJson(array $data = null, $negate = false)
...
+     * Assert that the response contains the given JSON keys.
+     *
+     * Keys can be formatted with dot notation, or as nested arrays:
+     *   $keys = ['key1', 'key2' => ['subkey1'], 'key3.subkey2']
+     *
+     * Would match this json reponse:
+     *   {"key1": "data", "key2": {"subkey1": "data"}, "key3": {"subkey2": "data"}}
+     *
+     * @param  array|null  $data
+     * @param  bool|false  $strict
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: bool|false is wrong because false is a subset of bool

Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: it should just be bool

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 2]
```diff
@@ -528,6 +528,82 @@ public function seeJson(array $data = null, $negate = false)
...
+        }
+
+        return $this->seeJsonKeysContain($data, $strict, true);
+    }
+
+    /**
+     * Assert that the response contains the given JSON keys.
+     *
+     * @param  array|null  $data
+     * @param  bool|false  $strict
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: same

File: src/Illuminate/Foundation/Testing/CrawlerTrait.php [hunk 3]
```diff
@@ -528,6 +528,82 @@ public function seeJson(array $data = null, $negate = false)
...
+        }
+
+        return $this->seeJsonKeysContain($data, $strict, false);
+    }
+
+    /**
+     * Assert that the response does not contain the given JSON keys.
+     *
+     * @param  array|null  $data
+     * @param  bool|false  $strict
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/CrawlerTrait.php]: same

File: src/Illuminate/Support/helpers.php [hunk 4]
```diff
@@ -772,3 +772,76 @@ function with($object)
...
+     *   ["key1", "key2", "key3"]
+     *
+     * array_flatten_keys($array, true, true):
+     *   ["key1", "key2", "key2.subkey1", "key3", "key3.subkey2"]
+     *
+     * array_flatten_keys($array, true, false):
+     *   ["key1", "key2", "subkey1", "key2.subkey1", "key3", "subkey2", "key3.subkey2"]
+     *
+     * @param  array  $array
+     * @param  bool|false  $deep
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: same

File: src/Illuminate/Support/helpers.php [hunk 5]
```diff
@@ -772,3 +772,76 @@ function with($object)
...
+     * @param  bool|false  $strict
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: same

**Comment by samlev** [comment]
Minor refactor - squashed to one commit

**Comment by taylorotwell** [comment]
Seems you could do this easier by just converting the JSON to an array and using array_has?

**Comment by samlev** [comment]
I'm not sure how that would handle the 'non-strict' version. Strict version would work fine like that.

**Comment by samlev** [comment]
@taylorotwell Ping on this - 'non-strict' wouldn't work with array_has (from my reading, at least).

**Comment by taylorotwell** [comment]
Going to hold off on adding this one to core.

**Pull request closed without merging**